### PR TITLE
games-engines/love: Version bump

### PR DESCRIPTION
**Pull request by BetaRays**
Added versions 0.10.2, 11.0 and 11.1, and updated the 9999 ebuild.

Closes: https://bugs.gentoo.org/640802
Closes: https://bugs.gentoo.org/646256
Package-Manager: Portage-2.3.24, Repoman-2.3.6

I am opening this pull request because #8058 got closed.

### CODE REVIEW

File: games-engines/love/love-0.10.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	fi
+}
+
+src_configure() {
+	econf --with-lua=$(usex luajit luajit lua)
+}
+
+pkg_postinst() {
+	gnome2_icon_cache_update
+	xdg_desktop_database_update
```
Comment (mgorny) [inline games-engines/love/love-0.10.2.ebuild]: You're missing xdg-utils inherit.

**Comment by BetaRays** [comment]
@mgorny I've updated the pull request with the changes you requested.

**Comment by Zlogene** [comment]
Could you please squash it?

**Comment by BetaRays** [comment]
@Zlogene @mgorny asked me to make separate commits for each version. But if you think it's cleaner I can do it.

**Comment by mgorny** [comment]
The commit boundaries are fine. I just need to test it and will merge it afterwards.

**Comment by mgorny** [comment]
Good enough for me. Few worthwhile goals for the future:
1. Remove `.la` files.
2. Look into slotting it, given that each version is incompatible.
3. Find some games we could use to actually test the particular versions.
4. Update dependencies in revdeps since AFAICS no game in ::gentoo works with the new versions.

**Comment by BetaRays** [comment]
@mgorny I updated the PR again.

File: games-arcade/orthorobot/orthorobot-0-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+DESCRIPTION="Perspective based puzzle game, where you flatten the view to move across gaps"
+HOMEPAGE="http://stabyourself.net/orthorobot/"
+SRC_URI="http://stabyourself.net/dl.php?file=${PN}/${PN}-source.zip -> ${P}.zip"
+
+LICENSE="CC-BY-NC-ND-3.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+LVSLOT="0.8"
+RDEPEND=">=games-engines/love-0.8.0:${LVSLOT}"
```
Comment (mgorny) [inline games-arcade/orthorobot/orthorobot-0-r2.ebuild]: To be honest, I don't think the `>=` dep makes sense given that the slot can't contain an older version, by definition.

File: games-action/trosh/trosh-20-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+SRC_URI="http://stabyourself.net/dl.php?file=trosh/trosh-linux.zip -> ${P}.zip
+	https://dev.gentoo.org/~hasufell/distfiles/${PN}.png"
+
+LICENSE="WTFPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+LVSLOT="0.8"
+RDEPEND="
+	>=games-engines/love-0.8.0:${LVSLOT}
```
Comment (mgorny) [inline games-action/trosh/trosh-20-r2.ebuild]: Likewise. While at it, could you please fix indentation below?

File: games-engines/love/love-0.10.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+src_configure() {
+	econf --with-lua=$(usex luajit luajit lua)
+}
+
+src_install() {
+	default
+
+	find "${D}" -name '*.la' -delete || die
+	if [ "$SLOT" != "0" ]
```
Comment (mgorny) [inline games-engines/love/love-0.10.2.ebuild]: Please always use `[[ ... ]]` in ebuilds, always do `${SLOT}` (with `{...}`) and avoid unnecessary quoting inside `[[ ... ]]`.

File: games-engines/love/love-0.10.2.ebuild [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	then
+		mv "${ED}/usr/bin/${PN}" "${ED}/usr/bin/${PN}-${SLOT}" || die
+		mv "${ED}"/usr/share/applications/love{,"-$SLOT"}.desktop || die
+		sed -i "s|/usr/bin/love|/usr/bin/love-$SLOT|" "${ED}/usr/share/applications/love-$SLOT.desktop" || die
+		rm "${ED}"/usr/{lib64/liblove.so,share/{mime/packages/love.xml,pixmaps/love.svg,icons/hicolor/scalable/mimetypes/application-x-love-game.svg,man/man1/love.1}} || die
```
Comment (mgorny) [inline games-engines/love/love-0.10.2.ebuild]: You're leaving some empty directories here, aren't you? Also, can't you remove whole `share/mime` directory etc.?

**Comment by BetaRays** [comment]
@mgorny updated

**Comment by BetaRays** [comment]
I added a sed script expression to include the slot in the .desktop file's “Name”.

**Pull request closed without merging**